### PR TITLE
Enable gocritic unlambda and wrapperfunc lints

### PR DESCRIPTION
**Pull request by camdencheek**
I'm planning to enable the `gocritic` lints in smaller PRs since there are many of them, and I'm not sure we want/need all of them. 

Progresses #18720 

Lints enabled:
- `unlambda` catches unnecessary wrapping of functions in anonymous functions
- `wrapperfunc` catches uses of stdlib functions with special args when there is a more specific function that does the same thing more explicitly. 

The only case `wrapperfunc` caught was `strings.Replace(..., -1) -> strings.ReplaceAll(...)`. Seemed worth changing since `ReplaceAll` is more explicit and readable than the magic -1, so I used comby (another great use case @rvantonder) to change it across the codebase with the following command:
```bash
comby -in-place 'strings.Replace(:[1],:[2],:[3], -1)' 'strings.ReplaceAll(:[1],:[2],:[3])' .go
```

### CODE REVIEW

**Review by eseliger — APPROVED** [review]

File: schema/stringdata.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func main() {
 }
 
 func backtickStringLiteral(data string) string {
-	return "`" + strings.Replace(data, "`", "` + \"`\" + `", -1) + "`"
+	return "`" + strings.ReplaceAll(data, "`", "` + \"`\" + `") + "`"
 }
```
Comment (rvantonder) [inline schema/stringdata.go:53]: We need to not let this affect generated files like `stringdata.go`

File: .golangci.enforced.yml [hunk 2]
```diff
@@ -16,16 +16,28 @@ linters:
...
+
+linters-settings:
+  gocritic:
+    enabled-checks:
+      - unlambda
+      - wrapperFunc
 
 issues:
   exclude-rules:
+    # Excluded generated bindata.go from gocritic lints
```
Comment (rvantonder) [inline .golangci.enforced.yml:29]: Needs to also exclude `stringdata.go` and others (not sure what the whole set is)

**Review by rvantonder — APPROVED** [review]
Needs minor fix up--double check excluding `stringdata.go` or things in `schema`.

File: schema/stringdata.go (hunk 1 above)
Comment (camdencheek) [reply schema/stringdata.go:53]: It doesn't look like `stringdata.go` is a generated file to me. It's a tool that generates files, so it seems to me that this one should be subject to the change

Comment (rvantonder) [reply schema/stringdata.go:53]: Oof sorry--I confused this with https://sourcegraph.com/github.com/sourcegraph/sourcegraph@e07f9a88276557d2e39065d8d9076e99e714a75e/-/blob/schema/settings_stringdata.go.

**Pull request merged**